src/share/vm/runtime/thread.cpp

```diff
@@ -4299,6 +4299,19 @@ JavaThread* Threads::find_java_thread_from_java_tid(jlong java_tid) {
       java_thread = thread;
```

Comment:
if the java_thread is found here, do we still need to search it in coroutine list?

Reply:
no. there is a break in this loop.
![Uploading image.png…]()

Reply:
for (JavaThread* thread = Threads::first(); thread != NULL; thread = thread->next()) {
    oop tobj = thread->threadObj();
    if (!thread->is_exiting() &&
        tobj != NULL &&
        java_tid == java_lang_Thread::thread_id(tobj)) {
      java_thread = thread;
      break;
    }